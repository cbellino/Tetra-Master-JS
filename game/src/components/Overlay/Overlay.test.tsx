import { shallow } from "enzyme";
import * as React from "react";

import { EnhancedOverlay, Overlay } from "./index";

const shallowUntil = component =>
  shallow(component)
    // @ts-ignore
    .until(Overlay)
    .dive();

describe("Overlay", () => {
  it("should display the children", () => {
    const wrapper = shallowUntil(<EnhancedOverlay>My content</EnhancedOverlay>);

    expect(wrapper.contains("My content")).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a className of 'overlay'", () => {
    const wrapper = shallowUntil(<EnhancedOverlay>My content</EnhancedOverlay>);

    expect(wrapper.hasClass("overlay")).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should react to mouse events", () => {
    const onClick = jest.fn();
    const wrapper = shallowUntil(<EnhancedOverlay onClick={onClick} />);

    wrapper.simulate("click");
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith();

    expect(wrapper).toMatchSnapshot();
  });
});
