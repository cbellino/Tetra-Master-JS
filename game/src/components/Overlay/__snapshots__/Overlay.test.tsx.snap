// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should display the children 1`] = `
<div
  className="overlay"
  onClick={[Function]}
  role="button"
>
  My content
</div>
`;

exports[`Overlay should have a className of 'overlay' 1`] = `
<div
  className="overlay"
  onClick={[Function]}
  role="button"
>
  My content
</div>
`;

exports[`Overlay should react to mouse events 1`] = `
<div
  className="overlay"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  role="button"
/>
`;
